### PR TITLE
Updated go marathon

### PR DESCRIPTION
**Pull request by Raffo**
I updated the version of go-marathon used and had to change some parts of the code to reflect the many breaking changes in it.

### CODE REVIEW

**Comment by Raffo** [comment]
@szuecs @njuettner @skoenig Please review.

File: backend/marathon.go [hunk 1]
```diff
@@ -185,44 +185,46 @@ func (mb *MarathonBackend) Deploy(cr *CreateRequest) (string, error) {
...
-	httpHealth := marathon.HealthCheck{Protocol: "HTTP", Path: "/health", GracePeriodSeconds: 3, IntervalSeconds: 10, MaxConsecutiveFailures: 10}
-	cmdHealth := marathon.HealthCheck{Protocol: "COMMAND", Command: &marathon.Command{Value: "curl -f -X GET  http://$HOST:$PORT0/health"}, MaxConsecutiveFailures: 10}
-	checks = append(checks, &httpHealth)
-	checks = append(checks, &cmdHealth)
+	//checks := make([]*marathon.HealthCheck, 0, 2)
+	//health := "/health"
+	// httpHealth := marathon.HealthCheck{Protocol: "HTTP", Path: &health, GracePeriodSeconds: 3, IntervalSeconds: 10, MaxConsecutiveFailures: 10}
+	// cmdHealth := marathon.HealthCheck{Protocol: "COMMAND", Command: &marathon.Command{Value: "curl -f -X GET  http://$HOST:$PORT0/health"}, MaxConsecutiveFailures: 10}
+	// checks = append(checks, &httpHealth)
+	// checks = append(checks, &cmdHealth)
```
Comment (szuecs) [inline backend/marathon.go]: commented code? Please stash it and save it in your local tree

File: backend/marathon.go [hunk 2]
```diff
@@ -160,9 +160,9 @@ func (mb *MarathonBackend) GetApp(req *ArtifactRequest) (*Artifact, error) {
 		Labels:            application.Labels,
 		Env:               application.Env,
 		RunningReplicas:   replicas,
-		RequestedReplicas: application.Instances,
+		RequestedReplicas: *application.Instances,
 		CPUS:              application.CPUs,
```
Comment (szuecs) [inline backend/marathon.go]: Why application.CPUs is not a pointer, but application.Mem ?

File: vendor/github.com/donovanhide/eventsource/encoder.go [hunk 3]
```diff
@@ -17,28 +18,41 @@ var (
 	}
 )
 
-type encoder struct {
-	w io.Writer
+// An Encoder is capable of writing Events to a stream. Optionally
```
Comment (szuecs) [inline vendor/github.com/donovanhide/eventsource/encoder.go]: To be godoc compliant you should start with the name of the function.

**Comment by szuecs** [comment]
PR looks fine to me, only these 3 small things should be done

File: backend/marathon.go (hunk 2 above)
Comment (Raffo) [inline backend/marathon.go]: Well, ask the go-marathon developers :-D I missed track of the development of the project and for some reason they messed up with pointers. I don't understand the reason behind the changes.

File: backend/marathon.go (hunk 1 above)
Comment (Raffo) [inline backend/marathon.go]: I can drop it as it is actually dead code.

File: vendor/github.com/donovanhide/eventsource/encoder.go (hunk 3 above)
Comment (Raffo) [inline vendor/github.com/donovanhide/eventsource/encoder.go]: It's in /vendor, it is not my code.

**Comment by njuettner** [comment]
PR looks good

**Comment by Raffo** [comment]
This were addressed, merging the PR.

**Pull request merged**